Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1516,10 +1516,59 @@ extension BrowserViewController: TabToolbarDelegate, PhotonActionSheetProtocol {
         showTabTray()
     }
 
+    func getTabToolbarLongPressActions() -> [PhotonActionSheetItem] {
+        var count = 1
+        let infinity = "\u{221E}"
+        if let selectedTab = tabManager.selectedTab {
+            count = selectedTab.isPrivate ? tabManager.normalTabs.count : tabManager.privateTabs.count
+        }
+        let tabCount = (count < 100) ? count.description : infinity
+
+        let privateBrowsingMode = PhotonActionSheetItem(title: Strings.privateBrowsingModeTitle, iconString: "nav-tabcounter", iconType: .TabsButton, tabCount: tabCount) { action in
+            if let tab = self.tabManager.selectedTab {
+                if self.tabManager.switchTabMode(tab) {
+                    let shouldFocusLocationField = NewTabAccessors.getNewTabPage(self.profile.prefs) == .blankPage
+                    self.openBlankNewTab(focusLocationField: shouldFocusLocationField, isPrivate: true)
+                }
+            }}
+        let normalBrowsingMode = PhotonActionSheetItem(title: Strings.normalBrowsingModeTitle, iconString: "nav-tabcounter", iconType: .TabsButton, tabCount: tabCount) { action in
+            if let tab = self.tabManager.selectedTab {
+                _ = self.tabManager.switchTabMode(tab)
+            }}
+        if let tab = self.tabManager.selectedTab {
+            return tab.isPrivate ? [normalBrowsingMode] : [privateBrowsingMode]
+        }
+        return [privateBrowsingMode]
+    }
+    func getMoreTabToolbarLongPressActions() -> [PhotonActionSheetItem] {
+        let newTab = PhotonActionSheetItem(title: Strings.NewTabTitle, iconString: "quick_action_new_tab", iconType: .Image) { action in
+            let shouldFocusLocationField = NewTabAccessors.getNewTabPage(self.profile.prefs) == .blankPage
+            self.openBlankNewTab(focusLocationField: shouldFocusLocationField, isPrivate: false)}
+        let newPrivateTab = PhotonActionSheetItem(title: Strings.NewPrivateTabTitle, iconString: "quick_action_new_tab", iconType: .Image) { action in
+            let shouldFocusLocationField = NewTabAccessors.getNewTabPage(self.profile.prefs) == .blankPage
+            self.openBlankNewTab(focusLocationField: shouldFocusLocationField, isPrivate: true)}
+        let closeTab = PhotonActionSheetItem(title: Strings.CloseTabTitle, iconString: "tab_close", iconType: .Image) { action in
+            if let tab = self.tabManager.selectedTab {
+                self.tabManager.removeTab(tab)
```

Comment:
This needs to be  self.tabManager.removeTabAndUpdateSelectedIndex(tab),
otherwise this patch is great!